diff -up kruler-19.08.2/klineal.cpp.omv~ kruler-19.08.2/klineal.cpp
--- kruler-19.08.2/klineal.cpp.omv~	2019-10-15 23:55:16.388096196 +0200
+++ kruler-19.08.2/klineal.cpp	2019-10-16 00:01:40.906981099 +0200
@@ -710,7 +710,7 @@ void KLineal::mouseReleaseEvent( QMouseE
 
 void KLineal::wheelEvent( QWheelEvent *e )
 {
-  int numDegrees = e->delta() / 8;
+  int numDegrees = e->angleDelta().manhattanLength() / 8;
   int numSteps = numDegrees / 15;
 
   // changing offset
@@ -761,8 +761,7 @@ void KLineal::drawScale( QPainter &paint
       }
 
       if ( digit % 100 == 0 && digit != 0 ) {
-        QString units;
-        units.sprintf( "%d", digit );
+        QString units = QString::asprintf( "%d", digit );
         drawScaleText( painter, x, units );
       }
 
@@ -776,9 +775,8 @@ void KLineal::drawScale( QPainter &paint
       int x = (int)( i * step );
 
       if ( i % 10 == 0 && i != 0 && i != 100 ) {
-        QString units;
         int value = mLeftToRight ? i : ( 100 - i );
-        units.sprintf( "%d%%", value );
+        QString units = QString::asprintf( "%d%%", value );
         drawScaleText( painter, x, units );
         len = MEDIUM2_TICK_SIZE;
       } else {
